bindings/go/src/examples/perftest/perftest.go

```diff
@@ -312,45 +301,46 @@ func clientStart() error {
 		return err
 	}
 
-	perfTest.perThreadWorkers = make([]*UcpWorker, perfTestParams.numThreads)
-	perfTest.eps = make([]*UcpEp, perfTestParams.numThreads)
-	for i := 0; i < int(perfTestParams.numThreads); i += 1 {
-		initWorker(i)
-		if err := clientConnectWorker(i); err != nil {
-			return err
-		}
+	initWorker()
+	if err := clientConnectWorker(); err != nil {
+		return err
 	}
 
-	var totalDuration time.Duration = 0
+	var start time.Time
+	perfTest.quit = make(chan struct{})
+	perfTest.numCompletedRequests = int32(-perfTestParams.warmUpIter)
 	printHeader()
-	for i := -int(perfTestParams.warmUpIter); i < int(perfTestParams.numIterations); i += 1 {
-		perfTest.wg.Add(int(perfTestParams.numThreads))
-		for t := uint(0); t < perfTestParams.numThreads; t += 1 {
-			go clientThreadDoIter(i, t)
+	go printStatistics()
```

Comment:
seems this threads runs polling in infinite loop, would it consume extra CPU core?

Reply:
ticker.C is blocking